api/src/services/users.ts

```diff
@@ -278,22 +293,28 @@ export class UsersService extends ItemsService {
 
 		const user = await this.knex.select('id', 'status').from('directus_users').where({ email }).first();
 
-		if (!user || user.status !== 'invited') {
+		if (user?.status !== 'invited') {
 			throw new InvalidPayloadException(`Email address ${email} hasn't been invited.`);
 		}
 
-		const passwordHashed = await argon2.hash(password);
-
-		await this.knex('directus_users').update({ password: passwordHashed, status: 'active' }).where({ id: user.id });
+		const service = new UsersService({
```

Comment:
This should use the `UserService.updateOne` to ensure the password meets security requirement.

Reply:
@rijkvanzanten This is the part that relates to #8163. We create a new instance of `UserService` to avoid any accountability protections that would prevent an unauthorized users updating the user record.

TODO: I should probably add a code comment explaining that.